controllers/servers/host_definer/kubernetes_manager/manager.py

```diff
@@ -0,0 +1,265 @@
+
+    def _get_host_definitions_api(self):
+        return self.dynamic_client.resources.get(api_version=settings.CSI_IBM_API_VERSION,
+                                                 kind=settings.HOST_DEFINITION_KIND)
+
+    def _get_csi_nodes_with_driver(self):
+        csi_nodes_with_driver = []
+        csi_nodes = self._get_csi_nodes()
+        for csi_node in csi_nodes:
+            if self._is_csi_node_has_driver(csi_node):
+                csi_nodes_with_driver.append(self._get_csi_node_object(csi_node))
+        return csi_nodes_with_driver
+
+    def _get_csi_nodes(self):
+        try:
+            return self.csi_nodes_api.get().items
+        except ApiException as ex:
+            logger.error(messages.FAILED_TO_GET_CSI_NODES.format(ex.body))
+            return []
+
+    def _get_nodes(self):
+        try:
+            return self.core_api.list_node().items
+        except ApiException as ex:
+            logger.error(messages.FAILED_TO_GET_NODES.format(ex.body))
+            return []
+
+    def _get_storage_classes(self):
+        try:
+            return self.storage_api.list_storage_class().items
+        except ApiException as ex:
+            logger.error(messages.FAILED_TO_GET_STORAGE_CLASSES.format(ex.body))
+            return []
+
+    def _is_csi_node_has_driver(self, csi_node):
+        if csi_node.spec.drivers:
+            for driver in csi_node.spec.drivers:
+                return driver.name == settings.IBM_BLOCK_CSI_PROVISIONER_NAME
+        return False
+
+    def _get_csi_node(self, node_name):
+        try:
+            csi_node = self.csi_nodes_api.get(name=node_name)
+            return self._get_csi_node_object(csi_node)
+        except ApiException as ex:
+            if ex.status != 404:
+                logger.error(messages.CSI_NODE_DOES_NOT_EXIST.format(node_name))
+            else:
+                logger.error(messages.FAILED_TO_GET_CSI_NODE.format(node_name, ex.body))
+            return CsiNode()
+
+    def _get_csi_node_object(self, csi_node):
+        csi_node_obj = CsiNode()
+        csi_node_obj.name = csi_node.metadata.name
+        csi_node_obj.node_id = self._get_node_id_from_csi_node(csi_node)
+        return csi_node_obj
+
+    def _get_node_id_from_csi_node(self, csi_node):
+        if csi_node.spec.drivers:
+            for driver in csi_node.spec.drivers:
+                if driver.name == settings.IBM_BLOCK_CSI_PROVISIONER_NAME:
+                    return driver.nodeID
+        return None
+
+    def _get_host_definition(self, node_name, secret):
+        try:
+            host_definitions = self._get_host_definitions()
+            for host_definition in host_definitions:
+                host_definition_obj = self._get_host_definition_object(host_definition)
+                if self._is_host_definition_matches(host_definition_obj, node_name, secret):
+                    return (host_definition_obj, 200)
+            return ('', 404)
```

Comment:
"empty" object is better to be None than empty str.. IMO

Reply:
plus 
![image](https://user-images.githubusercontent.com/33412258/182799050-d72c0b30-ffb8-4cac-8047-1faa702fc6cc.png)